src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter(person => person.sex === 'f'
+      && people.some(kid => kid.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const avgAge = women.reduce(
+    (prev, item) => prev + (item.died - item.born), 0) / women.length;
+
+  return avgAge;
```

Comment:
You could create a separate function for this, as you have similar in all functions